### ISSUE TITLE
Test build of Elmer as Flatpak now ready for testing.

### ISSUE THREAD

**Issue by kevinsmia1939** [ElmerCSC/elmerfem#324]
Hi,
I package Elmer as Flatpak for Linux.
https://www.flatpak.org/
https://flathub.org/home

Elmer's precompiled binary is available as a Flatpak. Any system that supports Flatpak should be able to run Elmer this way. Install Flatpak as directed here.
https://flatpak.org/setup/

To install this testing build of Elmer.
`flatpak install --user https://dl.flathub.org/build-repo/83307/fi.csc.Elmer.flatpakref`

The pull request for adding Elmer to Flathub and test build is here: https://github.com/flathub/flathub/pull/2992

Ideally, I would like any Elmer dev to become the maintainer of the Flathub repo, but I will do most of the Flatpak maintenance work.

At the moment, MUMPS and espreso is not include because I could not make Elmer build with it.
Let me know what you think.

**Comment by Taack** [comment]
Hi, I would like to test, but I get an error 404:
```bash
$ flatpak install --user https://dl.flathub.org/build-repo/83307/fi.csc.Elmer.flatpakref
error: Can't load uri https://dl.flathub.org/build-repo/83307/fi.csc.Elmer.flatpakref: Server returned status 404
```
Regards

**Comment by kevinsmia1939** [comment]
@Taack 
I have been patching up Elmer 9 to support newer dependencies atm, but there is quite a lot and will take some time. I wish elmer would release new version soon.

**Comment by kevinsmia1939** [comment]
@Taack 
Here is the new one
`flatpak install --user https://dl.flathub.org/build-repo/119888/fi.csc.Elmer.flatpakref`

**Comment by Taack** [comment]
@kevinsmia1939 it says the same error message, maybe it is a temporary link... 

Anyway, I could install ElmerFEM 9 on my machines. It works, but I noticed it depends on Qt4 so I imagine it is not an easy task to build a package... I will check later flatpak version you provide.

Regards

**Comment by kevinsmia1939** [comment]
> @kevinsmia1939 it says the same error message, maybe it is a temporary link...
> 
> Anyway, I could install ElmerFEM 9 on my machines. It works, but I noticed it depends on Qt4 so I imagine it is not an easy task to build a package... I will check later flatpak version you provide.
> 
> Regards

Yeah, it does not last more than a few days. I can trigger new build again.

**Comment by kevinsmia1939** [comment]
@Taack 
`flatpak install --user https://dl.flathub.org/build-repo/120817/fi.csc.Elmer.flatpakref`

**Comment by Taack** [comment]
@kevinsmia1939 

I could try this one, as far as I can tell, it works as expected. Regards.

**Comment by Ombrezz** [comment]
Are you still trying to get this on Flathub? If not, could you provide a permanent build somewhere?

**Comment by kevinsmia1939** [comment]
> Are you still trying to get this on Flathub? If not, could you provide a permanent build somewhere?

I am trying to publish to Flathub. But need someone to test it. Will rebuild it for testing again soon

**Comment by NN708** [comment]
I'm interested in the Flatpak build of Elmer but noticed that https://github.com/flathub/flathub/pull/2992 has been closed. Are there any issues that need to be resolved before Elmer can be published on Flathub?

**Comment by kevinsmia1939** [comment]
> I'm interested in the Flatpak build of Elmer but noticed that [flathub/flathub#2992](https://github.com/flathub/flathub/pull/2992) has been closed. Are there any issues that need to be resolved before Elmer can be published on Flathub?

I prefer if someone could test it out for me but there is no interest. If you want to test it out I will reopen the request.

**Comment by NN708** [comment]
I'd be happy to test the Flatpak build and help troubleshoot any issues that might come up.

**Comment by NN708** [comment]
Any news on this?

**Comment by kevinsmia1939** [comment]
> Any news on this?

Hi @NN708 , ElmerFEM has been added to Flathub. But I would like you to give it a test nonetheless. As of this comment, ElmerFEM has not been publish on Flathub (https://flathub.org) yet, but it should be soon in a few hour.
If you found any issue, please report it to https://github.com/flathub/fi.csc.Elmer/issues

**Comment by NN708** [comment]
@kevinsmia1939 Thanks so much for your great work on this! I just tested it with a simple example, and everything runs smoothly overall.

One small thing: I noticed a few icons are missing. I've opened an issue here: flathub/fi.csc.Elmer#2
